### DISCUSSION TITLE
android webview not use Ikev2 protocal.

### DISCUSSION THREAD

**Discussion by minh-dai** [strongswan/strongswan#1905] [Q&A] [answered]
Hello guys.
I connected ikev2 to my app, and it's successful. I open chorme app and check ip, it show my VPN.
But when I open webview in my app, and check my IP, it always shows my ip, not VPN.
How can i fix that??
Thanks

**Comment by tobiasbrunner** [comment, marked as answer]
The app excludes itself from the VPN, see here: https://github.com/strongswan/strongswan/blob/190d8cbe1931ec57484d9bb451824a7fc57979bd/src/frontends/android/app/src/main/java/org/strongswan/android/logic/CharonVpnService.java#L1124-L1137

**Reply by minh-dai** [reply]
yes, so I commented on this line, and it's working now.
appHandling = SelectedAppsHandling.SELECTED_APPS_EXCLUDE;

**Comment by minh-dai** [comment]
@tobiasbrunner 
Sorry, but i need some help.
My BE uses a router and uses ikev2/ipsec PSK, not ikev2/ipsec MSCHAPv2.
I saw that “strongSwan VPN Client for Android” does not support PSK
Link: https://docs.strongswan.org/docs/5.9/os/androidVpnClient.html

i found that Google has a pull request update, but i don't kown is its working?
Link: https://github.com/sijav/react-native-ip-sec-vpn/pull/3

How can i connect with ikev2/ipsec PSK
